sys/arm64/arm64/pmap.c

```diff
@@ -5771,7 +5795,7 @@ pmap_advise(pmap_t pmap, vm_offset_t sva, vm_offset_t eva, int advice)
 					 * dirty now.
 					 */
 					m = PHYS_TO_VM_PAGE(oldl3 & ~ATTR_MASK);
-					vm_page_dirty(m);
+					pmap_page_dirty(pmap, oldl3, m);
```

Comment:
This seems suspicious; it's under a pmap_pte_dirty check above, no pmap_pte_capdirty check.

Reply:
pmap_page_dirty will do the pmap_pte_capdirty check before capdirtying the vm_page; I thought it might be beneficial to propagate capdirty state in addition to normal dirty state to the page at this point. Happy to revert if you think that would be better.